### PR TITLE
Add PolarDirections

### LINKED ISSUE #1258: Add PolarDirections
In https://github.com/JuliaReach/LazySets.jl/pull/1255 we add `SphericalDirections`  useful for 3d template directions. Using the polar angle we can similarly add a `PolarDirections` struct, that has an integer argument that accounts for the discretization of `φ`.

### PR DESCRIPTION
**Pull request by mforets**
Closes #1258.

### CODE REVIEW

File: src/Approximations/template_directions.jl [hunk 1]
```diff
@@ -330,3 +330,86 @@ end
...
+
+### Examples
+
+The integer passed as an argument is used to discretize ``φ``:
+
+```jldoctest spherical_directions
+julia> using LazySets.Approximations: PolarDirections
+
+julia> pd = PolarDirections(3)
+PolarDirections{Float64}(3, Array{Float64,1}[[1.0, 0.0], [-1.0, 1.22465e-16]])
```
Comment (schillic) [inline src/Approximations/template_directions.jl]: I count only two elements here :confused:

File: src/Approximations/template_directions.jl [hunk 2]
```diff
@@ -330,3 +330,86 @@ end
...
+PolarDirections(Nφ::Int) = PolarDirections{Float64}(Nφ)
+
+# common functions
+Base.eltype(::Type{PolarDirections{N}}) where {N} = Vector{N}
+Base.length(pd::PolarDirections) = length(pd.stack)
+dim(::PolarDirections) = 2
+
+@static if VERSION < v"0.7-"
+    @eval begin
+        Base.start(pd::PolarDirections) = sd[1]
```
Comment (schillic) [inline src/Approximations/template_directions.jl]:
```suggestion
        Base.start(pd::PolarDirections) = pd[1]
```

File: src/Approximations/template_directions.jl (hunk 1 above)
Comment (mforets) [reply src/Approximations/template_directions.jl]: With `Nφ = 3` you get `φ = 0` and `φ = pi`, since `φ = 2pi` is discarded.

Comment (schillic) [reply src/Approximations/template_directions.jl]: I think you should not consider this duplicate in the count.

Comment (mforets) [reply src/Approximations/template_directions.jl]: Which count?

Comment (schillic) [reply src/Approximations/template_directions.jl]: `Nθ`

Comment (mforets) [reply src/Approximations/template_directions.jl]: Oh, you'd like that `PolarDirections(k)` returns exactly `k` vectors -- yes we can change that behavior if we let `φ = Compat.range(N(0.0), N(2*pi), length=Nφ+1)`. I'm fine either way.

For `SphericalCoordinates` it's less clear -- the length is `(Nφ-1)(Nθ-2) + 2`.

File: src/Approximations/template_directions.jl [hunk 3]
```diff
@@ -349,7 +431,6 @@ Spherical directions representation.
 
 ### Fields
 
-- `Nθ`    -- length of the partition of the azimuthal angle
```
Comment (schillic) [inline src/Approximations/template_directions.jl]: Why did you remove this line?

File: src/Approximations/template_directions.jl (hunk 1 above)
Comment (schillic) [reply src/Approximations/template_directions.jl]: I did the change.

File: src/Approximations/template_directions.jl [hunk 4]
```diff
@@ -338,6 +338,88 @@ end
...
+# ==================================================
+
+"""
+    PolarDirections{N<:AbstractFloat} <: AbstractDirections{N}
+
+Polar directions representation.
+
+### Fields
+
+- `Nθ`    -- length of the partition of the azimuthal angle
```
Comment (mforets) [inline src/Approximations/template_directions.jl]:
```suggestion
- `Nφ`    -- length of the partition of the polar angle
```

File: src/Approximations/template_directions.jl [hunk 5]
```diff
@@ -338,6 +338,88 @@ end
...
+    Nφ::Int
+    stack::Vector{Vector{N}} # stores the polar directions
+
+    function PolarDirections{N}(Nφ::Int) where {N<:AbstractFloat}
+        if Nφ <= 0
+            throw(ArgumentError("Nφ = $Nφ is invalid; it shoud be at least 1"))
+        end
+        stack = Vector{Vector{N}}()
+        # discretization of the polar angle
+        φ = Compat.range(N(0.0), N(2*pi), length=Nφ+1)
```
Comment (mforets) [inline src/Approximations/template_directions.jl]:
```suggestion
        φ = range(N(0.0), N(2*pi), length=Nφ+1)
```

File: src/Approximations/template_directions.jl (hunk 3 above)
Comment (mforets) [reply src/Approximations/template_directions.jl]: if i removed `Nθ` in Spherical directions it was unintentional, probably with the merge conflict.. thx
